File: gfapy/line/comment/tags.py
import gfapy

class Tags:

  def set(self, fieldname, value):
    """Set the value of a field.

    The generic Line.set() method is overwritten for comments,
    in order to disallow tags.
    """
    if fieldname in ["content", "spacer"]:
      return super().set(fieldname, value)
    else:
      raise gfapy.RuntimeError("Tags of comment lines cannot be set")

File: lib/rgfa/line/comment/tags.rb
# Comment lines do not support tags.
# @tested_in api_comments
module RGFA::Line::Comment::Tags

  # Set the value of the comment or sp field
  #
  # @param fieldname [:comment, :sp] the name of the field to set
  # @raise [RGFA::RuntimeError] if +fieldname+ is not one of the above
  # @return [Object] +value+
  def set(fieldname, value)
    if [:comment, :sp].include?(fieldname.to_sym)
      super
    else
      raise RGFA::RuntimeError,
        "Tags of comment lines cannot be set"
    end
  end

  private

  def method_missing(m, *args, &block)
    raise NoMethodError,
      "undefined method `#{m}' for #{self.inspect}"
  end

end
